ext/lb/lb.go

```diff
@@ -483,3 +495,14 @@ func (l *LoadBalancer) isExposedContainer(id string) bool {
 	log().Debugf("container is monitored; triggering reload: id=%s", id)
 	return true
 }
+
+func (l *LoadBalancer) isServiceContainer(labels map[string]string) bool {
+	containerServiceName, labelled := labels[ext.InterlockExtServiceNameLabel]
```

Comment:
So I don't know that changing it to a `switch` would improve readability. The following isn't anymore readable to me.

```go
switch {
case (len(l.cfg.ServiceName) > 0 && containerServiceName == l.cfg.ServiceName) || (len(l.cfg.ServiceName) == 0 && !labelled):
	return true
default:
	false
}
```

But the point about readability is well taken. What about something like this?

```go
func (l *LoadBalancer) isServiceContainer(labels map[string]string) bool {
	serviceName, labelled := labels[ext.InterlockExtServiceNameLabel]

	switch {
	case l.serviceNameMatch(serviceName):
		return true
	case l.serviceNamelessMatch(serviceName, labelled):
		return true
	default:
		false
	}
}

func (l *LoadBalancer) serviceNameMatch(serviceName string) {
	return len(l.cfg.ServiceName) > 0 && serviceName == l.cfg.ServiceName
}

func (l *LoadBalancer) serviceNamelessMatch(serviceName string, labelled bool) {
	return len(l.cfg.ServiceName) == 0 && !labelled
}
```

Thoughts?